### PR TITLE
[CustomOp]Add cpp_extension  en doc

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

Others
### PR changes

Docs
### Describe

Add cpp_extension  en doc, zh doc: https://github.com/PaddlePaddle/Paddle/pull/31187

image

image

image

image

image

### CODE REVIEW

**Review by chenwhql — DISMISSED** [review]
LGTM

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 1]
```diff
@@ -41,24 +41,93 @@
...
-                    include_dirs=[],       # specific user-defined include dirs
-                    extra_compile_args=[]) # specific user-defined compiler arguments.
+    The interface is used to config the process of compiling customized operators,
+    mainly includes how to complile shared library, automatically generate python API 
+    and install it into site-package. It supports using customized operators directly with
+    ``import`` statement.
+
+    It encapsulates the python built-in ``setuptools.setup`` function and keeps arguments
+    and usage same as the native interface. Meanwhile, it hiddens Paddle inner framework
+    concecepts, such as necessary compiling flags, included paths of head files, and linking
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: concecepts -> concepts

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 2]
```diff
@@ -41,24 +41,93 @@
...
+    local machine should satisfy GCC >= 8.2. Otherwise, a fatal error will occur because of
+    ABI compatibility.
+
+    .. note::
+
+        1. Compiler ABI compatibility is forward compatible. On Linux platform, 
+           we recommend to use GCC 8.2 as soft linking condidate of ``/usr/bin/cc`` .
+        2. Using ``which cc`` to ensure location of ``cc`` and using ``cc --version`` 
+           to ensure linking GCC version on Linux.
+        3. Currenly we support Linux and Windows platfrom. MacOS is supporting...
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: Currenly -> Currently

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 3]
```diff
@@ -41,24 +41,93 @@
...
+
+        setup(
+            name='custom_op',  # name of package used by "import"
+            ext_modules=CppExtension(
+                sources=['relu_op.cc', 'tanh_op.cc']  # Support for compilation of multiple OPs
+            )
+        )
+
+
+    Applying compilation and installation by execting ``python setup.py install`` under source files directory.
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: execting -> executing

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 4]
```diff
@@ -539,48 +660,86 @@ def load(name,
...
+                                all basic headers are included implicitly from ``site-package/paddle/include`` .
+                                Default None.
+        build_directory(str, optional): Specify root directory path to put shared library file. If set None,
+                            it will use ``PADDLE_EXTENSION_DIR`` from os.environ. Use
+                            ``paddle.utils.cpp_extension.get_build_directory()`` to see the location.
+        interpreter(str, optional): Specify nterpreter path, supporting alias and full path.
+                           If set None, it will use `python` as default interpreter. If local environment contains
+                           more than one python interpreters and want to use new interpreter to apply compilation,
+                           please specify this parameter, such as ``python3.7`` .
+        verbose(bool, optional): whether to verbose compiled log information.
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py]: Default is False

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 5]
```diff
@@ -539,48 +660,86 @@ def load(name,
...
-        interpreter(str): alias or full interpreter path to specific which one to use if have installed multiple.
-                           If set None, will use `python` as default interpreter.
-        verbose(bool): whether to verbose compiled log information
+        extra_include_paths(list[str], optional): Specify additional include path used to search header files. By default
+                                all basic headers are included implicitly from ``site-package/paddle/include`` .
+                                Default None.
+        build_directory(str, optional): Specify root directory path to put shared library file. If set None,
+                            it will use ``PADDLE_EXTENSION_DIR`` from os.environ. Use
+                            ``paddle.utils.cpp_extension.get_build_directory()`` to see the location.
+        interpreter(str, optional): Specify nterpreter path, supporting alias and full path.
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py:734]: Default None

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 6]
```diff
@@ -539,48 +660,86 @@ def load(name,
...
-        build_directory(str): specific directory path to put shared library file. If set None,
-                            it will use `PADDLE_EXTENSION_DIR` from os.environ. Use
-                            `paddle.utils.cpp_extension.get_build_directory()` to see the location.
-        interpreter(str): alias or full interpreter path to specific which one to use if have installed multiple.
-                           If set None, will use `python` as default interpreter.
-        verbose(bool): whether to verbose compiled log information
+        extra_include_paths(list[str], optional): Specify additional include path used to search header files. By default
+                                all basic headers are included implicitly from ``site-package/paddle/include`` .
+                                Default None.
+        build_directory(str, optional): Specify root directory path to put shared library file. If set None,
```
Comment (TCChenlong) [inline python/paddle/utils/cpp_extension/cpp_extension.py:731]: Default is None

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py]: Thanks, done

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 2 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py]: Thanks, done

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 3 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py]: Thanks, done

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 5 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py:734]: Thanks, done

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 6 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py:731]: Thanks, done

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by zhwesky2010 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Pull request merged**